src/Sentry/Protocol/Span.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol
+{
+    // https://develop.sentry.dev/sdk/event-payloads/span
+    public class Span : ISpan, IJsonSerializable
+    {
+        public SentryId SpanId { get; }
+        public SentryId? ParentSpanId { get; }
+        public SentryId TraceId { get; set; }
+        public DateTimeOffset StartTimestamp { get; set; }
+        public DateTimeOffset EndTimestamp { get; set; }
+        public string Operation { get; }
+        public string? Description { get; set; }
+        public SpanStatus? Status { get; set; }
+        public bool IsSampled { get; set; }
+
+        private ConcurrentDictionary<string, string>? _tags;
+        public IReadOnlyDictionary<string, string> Tags => _tags ??= new ConcurrentDictionary<string, string>();
+
+        private ConcurrentDictionary<string, object>? _data;
+        public IReadOnlyDictionary<string, object> Data => _data ??= new ConcurrentDictionary<string, object>();
```

Comment:
Maybe Lazy<T> with `PublicationOnly`

Reply:
Then we can't do deserialization

Reply:
Sorry, I don't follow, what do you mean?

The current approach is racy though. As concurrent access to this will yield multiple threads creating their own collection and assigning to the field, which is not immediately visible to all threads because the field isn't `volatile`. Perhaps a double check lock or `Interlocked.Exchange` could solve the problem here but I think `Lazy<T> with `PublicationOnly` is the simplest.

The simplest is to just have the instance from the start of the object but since this field is likely not going to be used often, paying the price to allocating it on each span, which will have dozens of instances per transaction, we're better off just using lazy or other mechanism